CMakeLists.txt

```diff
@@ -35,1037 +28,139 @@ link_libraries(Threads::Threads)
-    UnitTests/tALASCA_FourierMotzkin.cpp
-    UnitTests/tALASCA_TautologyDeletion.cpp
-    UnitTests/tALASCA_IntegerFourierMotzkin.cpp
-    UnitTests/tALASCA_InequalityFactoring.cpp
-    UnitTests/tALASCA_Normalization.cpp
-    UnitTests/tALASCA_FloorElimination.cpp
-    UnitTests/tALASCA_CoherenceNormalization.cpp
-    UnitTests/tALASCA_Abstraction.cpp
-    UnitTests/tALASCA_Superposition.cpp
-    UnitTests/tALASCA_Coherence.cpp
-    UnitTests/tALASCA_EqFactoring.cpp
-    UnitTests/tALASCA_VariableElimination.cpp
-    UnitTests/tALASCA_TermFactoring.cpp
-    UnitTests/tALASCA_Demodulation.cpp
-    UnitTests/tALASCA_VIRAS.cpp
-    UnitTests/tALASCA_LaKbo.cpp
-    UnitTests/tPushUnaryMinus.cpp
-    UnitTests/tArithmeticSubtermGeneralization.cpp
-    UnitTests/tInterpretedFunctions.cpp
-    UnitTests/tRebalance.cpp
-    UnitTests/tDisagreement.cpp
-    UnitTests/tDynamicHeap.cpp
-    UnitTests/tInduction.cpp
-    UnitTests/tIntegerConstantType.cpp
-    UnitTests/tSATSolver.cpp
-    UnitTests/tArithCompare.cpp
-    UnitTests/tSyntaxSugar.cpp
-    UnitTests/tSkipList.cpp
-    UnitTests/tBinaryHeap.cpp
-    UnitTests/tSafeRecursion.cpp
-    UnitTests/tKBO.cpp
-    UnitTests/tQKbo.cpp
-    UnitTests/tSKIKBO.cpp
-    UnitTests/tLPO.cpp
-    UnitTests/tRatioKeeper.cpp
-    UnitTests/tOptionConstraints.cpp
-    UnitTests/tDHMultiset.cpp
-    UnitTests/tList.cpp
-    UnitTests/tBottomUpEvaluation.cpp
-    UnitTests/tCoproduct.cpp
-    UnitTests/tIterator.cpp
-    UnitTests/tOption.cpp
-    UnitTests/tStack.cpp
-    UnitTests/tSet.cpp
-    UnitTests/tSATSubsumptionResolution.cpp
-    UnitTests/tDeque.cpp
-    UnitTests/tTermAlgebra.cpp
-    UnitTests/tFunctionDefinitionHandler.cpp
-    UnitTests/tFunctionDefinitionRewriting.cpp
-    )
-source_group(unit_tests FILES ${UNIT_TESTS})
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/viras/src)
 
-set(UNIT_TESTS_Z3
-    UnitTests/tTheoryInstAndSimp.cpp
-    UnitTests/tZ3Interfacing.cpp
-    )
-source_group(unit_tests_z3 FILES ${UNIT_TESTS_Z3})
+################################################################
+# GMP
+################################################################
 
+# TODO git submodule?
 
-# also include forwards.hpp?
-set(VAMPIRE_SOURCES
-    ${VAMPIRE_DEBUG_SOURCES}
-    ${VAMPIRE_LIB_SOURCES}
-    ${VAMPIRE_LIB_SYS_SOURCES}
-    ${VAMPIRE_KERNEL_SOURCES}
-    ${VAMPIRE_INDEXING_SOURCES}
-    ${VAMPIRE_INFERENCE_SOURCES}
-    ${VAMPIRE_SAT_SOURCES}
-    ${VAMPIRE_DECISION_PROCEDURES_SOURCES}
-    ${VAMPIRE_SATURATION_SOURCES}
-    ${VAMPIRE_SHELL_SOURCES}
-    ${VAMPIRE_PARSE_SOURCES}
-    ${VAMPIRE_FINITEMODELBUILDING_SOURCES}
-    ${VAMPIRE_SMTCOMP_SOURCES}
-    ${VAMPIRE_MINISAT_SOURCES}
-    ${VAMPIRE_CADICAL_SOURCES}
-    ${VAMPIRE_CASC_SOURCES}
-    ${VAMPIRE_SAT_SUBSUMPTION_SOURCES}
-    Forwards.hpp
-    "${CMAKE_CURRENT_BINARY_DIR}/version.cpp"
-    )
+# TODO could this just be source files?
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/mini-gmp-6.3.0)
```

Comment:
I think yes they could be source files, but I think having them as an include dir makes more clear that we don't wanna touch them.